private/buf/cmd/buf/command/dep/internal/internal.go

```diff
@@ -114,6 +120,23 @@ func Prune(
 	if err != nil {
 		return err
 	}
+	// If keepUnused is set to true, we add the unused declared dependencies to depModuleKeys.
+	if keepUnused {
+		moduleKeyProvider, err := bufcli.NewModuleKeyProvider(container)
+		if err != nil {
+			return err
+		}
+		unusedModuleKeys, err := moduleKeyProvider.GetModuleKeysForModuleRefs(
+			ctx,
+			unusedModuleRefs,
+			workspaceDepManager.BufLockFileDigestType(),
+		)
+		if err != nil {
+			return err
+		}
+		depModuleKeys = append(depModuleKeys, unusedModuleKeys...)
```

Comment:
We should make sure that the union of `depModuleKeys` and  `unusedModuleKeys` are unique by `ModuleFullName`. Do we expect that `unusedModuleKeys` will never have overlap with `depModuleKeys`? If so, error if not unique, otherwise we need to dedupe.

Reply:
Yeah, I think our expectation is that they should not overlap, since `depModuleKeys` is pulled from what is used and `unusedModuleKeys` should only be things that have never been used, so agree, we should error here.